cli/pkg/cicd/initiator/vmclarity.go

```diff
@@ -0,0 +1,231 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package initiator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/cli/pkg/cicd/vminfoprovider"
+	cliutils "github.com/openclarity/vmclarity/cli/pkg/utils"
+	"github.com/openclarity/vmclarity/runtime_scan/pkg/utils"
+	"github.com/openclarity/vmclarity/shared/pkg/backendclient"
+	"github.com/openclarity/vmclarity/shared/pkg/families"
+)
+
+type VMClarityInitiator struct {
+	client             *backendclient.BackendClient
+	input              string
+	targetInfoType     string
+	scanConfigName     string
+	scanConfigID       string
+	scanConfigFamilies *models.ScanFamiliesConfig
+}
+
+func NewInitiator(
+	client *backendclient.BackendClient,
+	config *families.Config,
+	scanConfigID,
+	scanConfigName,
+	input,
+	inputType string) (*VMClarityInitiator, error) {
+	if client == nil {
+		return nil, errors.New("backend client must not be nil")
+	}
+	targetInfoType, err := getTargetInfoType(inputType)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get target type by inputType=%s: %v", inputType, err)
+	}
+	return &VMClarityInitiator{
+		client:             client,
+		input:              input,
+		targetInfoType:     targetInfoType,
+		scanConfigName:     scanConfigName,
+		scanConfigID:       scanConfigID,
+		scanConfigFamilies: cliutils.ConvertScanFamiliesConfigToAPIModel(config),
+	}, nil
+}
+
+func getTargetInfoType(inputType string) (string, error) {
+	switch inputType {
+	case "dir", "DIR":
+		return "DIRInfo", nil
+	case "vm", "VM":
+		return "VMInfo", nil
+	default:
+		return "", errors.New("unsupported target type")
+	}
+}
+
+func (i *VMClarityInitiator) InitResults(ctx context.Context) (string, string, error) {
+	targetID, err := i.createTarget(ctx)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to init results: %v", err)
```

Comment:
please fix the error message in all calls, they all have `failed to init results...`